src/toast/ops/gainscrambler.py

```diff
@@ -0,0 +1,124 @@
+# Copyright (c) 2015-2020 by the parties listed in the AUTHORS file.
+# All rights reserved.  Use of this source code is governed by
+# a BSD-style license that can be found in the LICENSE file.
+
+import re
+
+import traitlets
+
+from .operator import Operator
+
+from ..traits import trait_docs, Int, Float, Unicode
+
+from .. import rng
+
+from ..utils import Logger
+
+from ..timing import function_timer
+
+
+@trait_docs
+class GainScrambler(Operator):
+    """Apply random gain errors to detector data.
+
+    This operator draws random gain errors from a given distribution and
+    applies them to the specified detectors.
+    """
+
+    # Class traits
+
+    API = Int(0, help="Internal interface version for this operator")
+
+    det_data = Unicode("signal", help="Observation detdata key apply the gain error to")
+
+    pattern = Unicode(
+        f".*",
+        allow_none=True,
+        help="Regex pattern to match against detector names. Only detectors that "
+        "match the pattern are scrambled.",
+    )
+    center = Float(1, allow_none=False, help="Gain distribution center")
+
+    sigma = Float(1e-3, allow_none=False, help="Gain distribution width")
+
+    realization = Int(0, allow_none=False, help="Realization index")
+
+    component = Int(0, allow_none=False, help="Component index for this simulation")
+
```

Comment:
This operator does not use any flags or data views.  There are some other operators that do this (e.g. `SimNoise`) because they need to generate a contiguous series of samples.  Ok to leave this as-is, but if we find a situation where we want to only scramble gains over specific intervals then we should add a view trait here.